src/constants.ts

```diff
@@ -51,3 +51,45 @@ export const CLASSIC_MODE_MYSTERY_ENCOUNTER_WAVES: [min: number, max: number] =
 
 /** Min - Max waves for mystery encounter in challenge mode. */
 export const CHALLENGE_MODE_MYSTERY_ENCOUNTER_WAVES: [min: number, max: number] = [10, 180];
+
+/**
+ * Spawn chance: (BASE_MYSTERY_ENCOUNTER_SPAWN_WEIGHT + WIGHT_INCREMENT_ON_SPAWN_MISS * <number of missed spawns>) / MYSTERY_ENCOUNTER_SPAWN_MAX_WEIGHT
+ */
+export const MYSTERY_ENCOUNTER_BASE_SPAWN_WEIGHT = 3;
+
+/**
+ * The divisor for determining ME spawns, defines the "maximum" weight required for a spawn
+ * If spawn_weight === MYSTERY_ENCOUNTER_SPAWN_MAX_WEIGHT, 100% chance to spawn a ME
+ */
+export const MYSTERY_ENCOUNTER_SPAWN_MAX_WEIGHT = 256;
+
+/**
+ * When an ME spawn roll fails, WEIGHT_INCREMENT_ON_SPAWN_MISS is added to future rolls for ME spawn checks.
+ * These values are cleared whenever the next ME spawns, and spawn weight returns to BASE_MYSTERY_ENCOUNTER_SPAWN_WEIGHT
```

Comment:
```suggestion
 * When an ME spawn roll fails, MYSTERY_ENCOUNTER_WEIGHT_INCREMENT_ON_SPAWN_MISS is added to future rolls for ME spawn checks.
 * These values are cleared whenever the next ME spawns, and spawn weight returns to MYSTERY_ENCOUNTER_BASE_SPAWN_WEIGHT
```

These are very long, I would've maybe used ME_ rather than MYSTERY_ENCOUNTER_ as a prefix for these, even if it's not the best practice